backend/tests/test_models_endpoints.py

```diff
@@ -1,24 +1,216 @@
+    "valid_substances,concs,expected_concs",
+    [(["H2"], {"H2": 10}, None), ([], {"H2": 10}, {}), (["H2"], {}, {"H2": 0.0})],
+)
+def test_dummy_model_only_valid_substances_are_present(
+    client, dummy_model, monkeypatch, valid_substances, concs, expected_concs
+):
+    monkeypatch.setattr(dummy_model, "valid_substances", valid_substances)
+
+    response = client.post(
+        f"/models/{dummy_model.model_id}/runs",
+        json={
+            "concs": concs,
+            "settings": {},
+        },
+    )
+    response.raise_for_status()
+    assert response.json() == {
+        "initialConcentrations": concs if expected_concs is None else expected_concs,
+        "finalConcentrations": concs if expected_concs is None else expected_concs,
+        "panels": [],
+    }
+
+
+class _DummyModelParameters(BaseParameters):
+    some_field: int = Parameter(0)
+
+
+class _DummyModelParametersMinMax(BaseParameters):
+    some_field: int = Parameter(0, min=-5, max=5)
+
+
+def test_dummy_has_correct_parameter_name(client, monkeypatch, dummy_model):
+    async def run(self):
+        return self.concentrations, JsonResult(data=self.parameters)
+
+    monkeypatch.setitem(
+        dummy_model.__annotations__, "parameters", _DummyModelParameters
+    )
+    monkeypatch.setattr(dummy_model, "run", run)
+
+    response = client.get_json("/models")
+
+    assert response == [
+        {
+            "accessError": None,
+            "displayName": "Dummy Model",
+            "modelId": "dummy",
+            "parameters": {
+                # Notice that it's "someField" and not "some_field"
+                "someField": {
+                    "__type__": "AcidwatchParameter",
+                    "choices": None,
+                    "convertibleUnit": None,
+                    "default": 0,
+                    "description": None,
+                    "label": None,
+                    "title": "Somefield",
+                    "type": "integer",
+                    "unit": None,
+                }
+            },
+            "validSubstances": [],
+        }
+    ]
+
+
+@pytest.mark.parametrize(
+    "parameters_class,input_parameters,expected",
+    [
+        pytest.param(
+            None,
+            {"someField": 123},
+            "Model dummy expected no parameters, got {'someField': 123}",
+            id="no params defined",
+        ),
+        pytest.param(
+            _DummyModelParameters,
+            {"someField": 123},
+            {"someField": 123},
+            id="valid camel case param",
+        ),
+        pytest.param(_DummyModelParameters, {}, {"someField": 0}, id="default value"),
+        pytest.param(
+            _DummyModelParameters,
+            {"some_field": 123},
+            {"someField": 123},
+            id="valid snake case param",
+        ),
+        pytest.param(
+            _DummyModelParameters,
+            {"someOtherField": 123},
+            "Extra inputs are not permitted",
```

Comment:
Test says "test_dummy_model_only_valid_parameters_are_present", but in this case we have invalid parameters I suppose..?

I think this is nice to include in the test.

I wonder though, I think the error message should include the name of the invalid parameter.